app/core/ai.py

```diff
@@ -58,101 +61,102 @@ def initialize_agent(aid):
+            raise HTTPException(status_code=404, detail="Agent not found")
+        except SQLAlchemyError as e:
+            # Handle other SQLAlchemy-related errors
+            logger.error(e)
+            raise HTTPException(status_code=500, detail=str(e))
+        # Initialize LLM.
+        llm = ChatOpenAI(model_name=agent.model, openai_api_key=config.openai_api_key)
+
+        # Load tools
+        tools: list[BaseTool] = []
+
+        # Configure CDP Agentkit Langchain Extension.
+        if agent.cdp_enabled:
+            values = {
+                "cdp_api_key_name": config.cdp_api_key_name,
+                "cdp_api_key_private_key": config.cdp_api_key_private_key,
+                "network_id": getattr(agent, "cdp_network_id", "base-sepolia"),
+            }
+            if agent.cdp_wallet_data:
+                # If there is a persisted agentic wallet, load it and pass to the CDP Agentkit Wrapper.
+                values["cdp_wallet_data"] = agent.cdp_wallet_data
+            agentkit = CdpAgentkitWrapper(**values)
+            # save the wallet after first create
+            if not agent.cdp_wallet_data:
+                agent.cdp_wallet_data = agentkit.export_wallet()
+                db.add(agent)
+                db.commit()
+            # Initialize CDP Agentkit Toolkit and get tools.
+            cdp_toolkit = CdpToolkit.from_cdp_agentkit_wrapper(agentkit)
+            tools.extend(cdp_toolkit.get_tools())
+        # Twitter skills
+
+        if agent.twitter_skills and agent.twitter_enabled and agent.twitter_config:
+            twitter_client = tweepy.Client(**agent.twitter_config)
+            for skill in agent.twitter_skills:
+                tools.append(get_twitter_skill(skill, twitter_client))
+
+        # Crestal skills
+        if agent.crestal_skills:
+            for skill in agent.crestal_skills:
+                tools.append(get_crestal_skill(skill))
+
+        # Common skills
+        if agent.common_skills:
+            for skill in agent.common_skills:
+                tools.append(get_common_skill(skill))
+
+        # Skill sets
+        if agent.skill_sets:
+            for skill_set, opts in agent.skill_sets.items():
+                tools.extend(get_skill_set(skill_set, opts))
+
+        # filter the duplicate tools
+        tools = list({tool.name: tool for tool in tools}.values())
+
+        # log all tools
+        for tool in tools:
+            logger.info(f"[{aid}] loaded tool: {tool.name}")
+
+        # Store buffered conversation history in memory.
+        memory = PostgresSaver(get_coon())
+        memory.setup()
+
+        prompt = ""
+        if agent.name:
+            prompt = f"Your name is {agent.name}. "
+
+        if agent.prompt:
+            prompt += agent.prompt
+        elif agent.cdp_enabled:
+            prompt += (
+                "You are a helpful agent that can interact onchain using the Coinbase Developer Platform AgentKit. "
```

Comment:
we should remove this a bit later to make it more extensible
essentially, each skill should have some default "system prompt" that safeguards things

Reply:
Each skill do have their prompt, they will pass to model too. This one only takes effect when the user has not provided a custom prompt and the CDP tool is enabled. We can come up with something else to replace it in future updates.